### PR TITLE
fix: unpublish preview for Multi-language

### LINKED ISSUE #86: multilang: preview does not work if lang param is used in query
Hey

Thanks for this plugin!

If I use a lang param in a page query like this : `allPosts(uid: $uid, lang: $lang) {...` 
The `lang` from the query wont be set because of this check `if (!params[key] && qs.has(key))`
here: https://github.com/birkir/gatsby-source-prismic-graphql/blob/master/packages/gatsby-source-prismic-graphql/src/components/WrapPage.tsx#L66

It does not matter if the languages share the same `uid` or not.

If I remove `!params[key]`, previews work for each language in my use case. 
But I didn't test this with other configurations.

Is the above check necessary?

Thanks

### LINKED ISSUE #126: Error when previewing unpublished pages on localhost that matches the pattern `article/:uid`
### Description

I want to preview an automatically generated page that matches the pattern `/article/:uid`. The page is unpublished and I am trying to access it on `localhost`. 

Previewing the same page on my staging site works fine, the page is accessible under `/article?uid=baz`. But when previewing it on localhost, I am routed to `/article/baz`. This results in an error since this page is not published.

### Steps to Reproduce

Add the following snippet in your `gatsby-config.js` (together with other settings): 

```javascript
pages: [
          {
            type: 'Page',
            match: '/:uid',
            path: '/page',
            component: require.resolve('./src/templates/page.js')
          },
          {
            type: 'Article',
            match: '/article/:uid',
            path: '/article',
            component: require.resolve('./src/templates/article.js')
          }
        ]

```

And in your `linkResolver.js`: 

```javascript
export const linkResolver = doc => {
  if (doc.type === 'article') {
    return `/article/${doc.uid}`
  }

  if (doc.type === 'page') {
    return `/${doc.uid}`
  }

  // Backup for all other types
  return '/'
}
```

**Expected behavior:**
Unpublished article pages should accessible for preview on localhost under `/article?uid=baz`.

**Actual behavior:** 

Unpublished article page is not accessible for preview on localhost.

### Versions
```
"gatsby": "2.18.11"
"prismic-reactjs": "1.1.2",
"gatsby-source-prismic-graphql": "3.4.0-beta.2"
```

### PR DESCRIPTION
**Pull request by Duaner**
**Problem**
When building the preview page for a specific Custom type, a lang is set statically in the context and not properly replaced by the one passed through the URL Path or QueryString.

Fixes birkir/gatsby-source-prismic-graphql#86

**Description**

We now use only queryString for unpublished documents since we had conflict in the router. In development, if you were using the DefaultLang option and tried to preview an unpublish doc, the 404 was always showing. We bypassed that by using queryString for all unpublished docs. If the doc is already published we redirect to a pretty URL.

Added a new utils to get QueryString as JSON

### Bug fixes
- fix previews in multi-lang project
### Breaking changes
- `path` is no longer used in `gatsby-config.js`; you should use `match` to define your pages path ; and you can use `previewPath` if you want to change the default route for unpublished document preview page

I also updated to the newest toolbar that should Fix birkir/gatsby-source-prismic-graphql#126

### CODE REVIEW

File: packages/gatsby-source-prismic-graphql/src/components/WrapPage.tsx [hunk 1]
```diff
@@ -50,17 +50,17 @@ export class WrapPage extends React.PureComponent<any, WrapPageState> {
...
 
-    const matchFn = pathToRegexp.match(get(this.props.pageContext, 'matchPath', ''), {
+    const matchFn = matchRegex(get(this.props.pageContext, 'matchPath', ''), {
       decode: decodeURIComponent,
     });
 
     const pathParams = (() => {
       const res = matchFn(get(this.props, 'location.pathname', ''));
-      return res.params;
+      return res === false ? '' : res.params;
```
Comment (arnaudlewis) [inline packages/gatsby-source-prismic-graphql/src/components/WrapPage.tsx]: pathParams is supposed to return an Object so your default value should not be ''

File: packages/gatsby-source-prismic-graphql/src/utils/parseQueryString.ts [hunk 2]
```diff
@@ -12,7 +12,19 @@ export function parseQueryString(qs: string, delimiter: string = '&'): Map<strin
 export function parseQueryStringAsJson(
   qs: string = '',
   delimiter: string = '&'
-): Map<string, string> {
+): { [key: string]: string } {
   const qsMap = parseQueryString(qs, delimiter);
-  return Object.fromEntries(qsMap);
+
+  console.log('qsMap', qsMap);
```
Comment (arnaudlewis) [inline packages/gatsby-source-prismic-graphql/src/utils/parseQueryString.ts]: remove this

**Review by arnaudlewis — CHANGES_REQUESTED** [review]

File: packages/gatsby-source-prismic-graphql/src/components/WrapPage.tsx (hunk 1 above)
Comment (Raph22) [reply packages/gatsby-source-prismic-graphql/src/components/WrapPage.tsx]: it was me as a dumb "fix" :p

Comment (Duaner) [reply packages/gatsby-source-prismic-graphql/src/components/WrapPage.tsx]: Done

File: packages/gatsby-source-prismic-graphql/src/utils/parseQueryString.ts (hunk 2 above)
Comment (Duaner) [reply packages/gatsby-source-prismic-graphql/src/utils/parseQueryString.ts]: Done

**Comment by MadsMadsDk** [comment]
Looking forward to this one getting merged.

**Comment by Duaner** [comment]
I just realized that there is still a problem there. When you use default lang, the URL for the preview page is properly set but doesn't load properly the preview component but the 404 page. 

I will see what i can do for this next week.

**Comment by Duaner** [comment]
I just realized that there is still a problem there. When you use default lang, the URL for the preview page is properly set but doesn't load properly the preview component but the 404 page. 

I will see what i can do for this next week.

**Comment by MadsMadsDk** [comment]
@Duaner Did you make any progress on this? :)

**Comment by jdwillemse** [comment]
@Duaner how can we help to get this merged?

**Comment by jdwillemse** [comment]
@veloce a Prettier check fails on your last commit

**Comment by ruslansavenok** [comment]
For anyone who needs a fix before this gets merged, just hardcode urls for each langauge in `pages: []` instead of using `:lang?`:

```
{
  type: 'Blog_post',
  match: '/it/blog/:uid',
  path: '/it/blog-preview',
  component: require.resolve('./src/templates/blog_post.js'),
  langs: ['it-it'],
},
{
  type: 'Blog_post',
  match: '/blog/:uid',
  path: '/blog-preview',
  component: require.resolve('./src/templates/blog_post.js'),
  langs: ['en-us'],
},
```

**Pull request merged**